python/triton/language/semantic.py

```diff
@@ -1352,11 +1333,11 @@ def assert_dtypes_valid(lhs_dtype, rhs_dtype, target):
         assert acc.type == ret_ty
 
     # max_num_imprecise_acc only applies to fp8 -> fp32 dot on sm_90
-    if not (_is_cuda(builder.target) and builder.target.capability == 90 and lhs.dtype.is_fp8() and rhs.dtype.is_fp8()
-            and ret_scalar_ty.is_fp32()):
-        max_num_imprecise_acc = 0
     if max_num_imprecise_acc is None:
-        max_num_imprecise_acc = 2**30
+        if lhs.dtype.is_fp8() and rhs.dtype.is_fp8():
+            max_num_imprecise_acc = builder.options.max_num_imprecise_acc_default
+        else:
+            max_num_imprecise_acc = 0
```

Comment:
I've just copied this part from upstream to make this function more upstream-like.
Currently we do not run this code, so nothing should change for us.